src/v/cluster/health_monitor_backend.h

```diff
@@ -128,6 +135,7 @@ class health_monitor_backend {
     ss::sharded<rpc::connection_cache>& _connections;
     ss::sharded<partition_manager>& _partition_manager;
     ss::sharded<raft::group_manager>& _raft_manager;
+    std::optional<health_node_cb_t> _node_cb;
```

Comment:
maybe more generic calback registration mechanism would be good here. Now there may be only one callback registered and it couples lifecycle of the calback registering object with health monitor backend. We have more generic callback implementations f.e. in `storage::log_manager`

Reply:
I've generalized this -- I was originally trying to avoid unnecessary generality, but the lifetime part makes this worthwhile.  I've done something similar to group_manager.